[menubar] 'MenuItem' is imported from both 'package:flutter/src/widgets/platform_menu_bar.dart' and 'package:menubar/src/menu_item.dart'

### Is this a bug in one of the plugins in this project?

- [X] I have read [the README section about feedback](https://github.com/google/flutter-desktop-embedding/blob/master/README.md#feedback).
- [X] My issue is about one of the plugins in this repository.

### Which plugin is this bug report about?

menubar

### Describe the bug

Im using Windows 11, After upgrading from Flutter Version 2 to latest version, i.e Flutter V 3.0.4 I'm getting this particular error 
```
/C:/src/flutter/.pub-cache/git/flutter-desktop-embedding-5c51870ced62a00e809ba4b81a846a052d241c9f/plugins/menubar/lib/src/menu_channel.dart(19,1): error GD26961E6: 'MenuItem' is imported from both 'package:flutter/src/widgets/platform_menu_bar.dart' and 'package:menubar/src/menu_item.dart'. [path\to\project\build\windows\flutter\flutter_assemble.vcxproj]

C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(245,5): error MSB8066: Custom build for 'path\to\project\build\windows\CMakeFiles\35bdcf3b678269ad2c9ea788c0dbc6c5\flutter_windows.dll.rule;path\to\project\build\windows\CMakeFiles\a0f4c5dbf216fa51231bbc5c8c02a604\flutter_assemble.rule' exited with code 1. [path\to\project\build\windows\flutter\flutter_assemble.vcxproj]
```

In pubspec.yaml I have added this dependencies
```
menubar:
    git:
      url: https://github.com/google/flutter-desktop-embedding.git
      path: plugins/menubar
      ref: 5c51870ced62a00e809ba4b81a846a052d241c9f
````

and In the code I'm accessing like this

```dart
import 'package:menubar/menubar.dart' as menubar;

  menubar.Submenu(
    label: 'Add',
    children: [
      menubar.MenuItem(
          label: "Home",
          onClicked: () {
          },
        )
    ]
  )
```

### Flutter doctor

```bash
[√] Flutter (Channel stable, 3.0.4, on Microsoft Windows [Version 10.0.21996.1], locale en-IN)
    • Flutter version 3.0.4 at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 85684f9300 (6 days ago), 2022-06-30 13:22:47 -0700
    • Engine revision 6ba2af10bb
    • Dart version 2.17.5
    • DevTools version 2.12.2

[√] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at C:\Users\nesan\AppData\Local\Android\sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.1.4)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.1.32407.343
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)

[√] VS Code (version 1.68.1)
    • VS Code at C:\Users\nesan\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.44.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.21996.1]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 103.0.5060.114
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 102.0.1245.44

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

--- a/plugins/menubar/lib/menubar.dart
+++ b/plugins/menubar/lib/menubar.dart
@@ -11,5 +11,5 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-export 'src/menu_item.dart';
+export 'src/native_menu_item.dart';
 export 'src/set_application_menu.dart';

--- a/plugins/menubar/lib/src/menu_channel.dart
+++ b/plugins/menubar/lib/src/menu_channel.dart
@@ -16,7 +16,7 @@ import 'dart:async';
 import 'package:flutter/services.dart';
 import 'package:flutter/widgets.dart';
 
-import 'menu_item.dart';
+import 'native_menu_item.dart';
 
 /// Whether or not the menu item is a divider, as a boolean. If true, no other
 /// The name of the plugin's platform channel.
@@ -113,7 +113,7 @@ class MenuChannel {
 
   /// Map from unique identifiers assigned by this class to the callbacks for
   /// those menu items.
-  final Map<int, MenuSelectedCallback> _selectionCallbacks = {};
+  final Map<int, VoidCallback> _selectionCallbacks = {};
 
   /// The ID to use the next time a menu item needs an ID assigned.
   int _nextMenuItemId = 1;
@@ -133,7 +133,7 @@ class MenuChannel {
   /// How exactly this is handled is subject to platform interpretation.
   /// For instance, special menus that are handled entirely on the native
   /// side might be added to the provided menus.
-  Future<Null> setMenu(List<Submenu> menus) async {
+  Future<Null> setMenu(List<NativeSubmenu> menus) async {
     try {
       _updateInProgress = true;
       await _platformChannel.invokeMethod(
@@ -150,7 +150,7 @@ class MenuChannel {
   /// As a side-effect, repopulates _selectionCallbacks with a mapping from
   /// the IDs assigned to any menu item with a selection handler to the
   /// callback that should be triggered.
-  List<dynamic> _channelRepresentationForMenus(List<Submenu> menus) {
+  List<dynamic> _channelRepresentationForMenus(List<NativeSubmenu> menus) {
     _selectionCallbacks.clear();
     _nextMenuItemId = 1;
 
@@ -160,22 +160,21 @@ class MenuChannel {
   /// Returns a representation of [item] suitable for passing over the
   /// platform channel to the native plugin.
   Map<String, dynamic> _channelRepresentationForMenuItem(
-      AbstractMenuItem item) {
+      AbstractNativeMenuItem item) {
     final representation = <String, dynamic>{};
-    if (item is MenuDivider) {
+    if (item is NativeMenuDivider) {
       representation[_kDividerKey] = true;
     } else {
       representation[_kLabelKey] = item.label;
-      if (item is Submenu) {
+      if (item is NativeSubmenu) {
         representation[_kChildrenKey] =
             _channelRepresentationForMenu(item.children);
-      } else if (item is MenuItem) {
-        final handler = item.onClicked;
-        if (handler != null) {
-          representation[_kIdKey] = _storeMenuCallback(handler);
-        }
-        if (!item.enabled) {
+      } else if (item is NativeMenuItem) {
+        final handler = item.onSelected;
+        if (handler == null) {
           representation[_kEnabledKey] = false;
+        } else {
+          representation[_kIdKey] = _storeMenuCallback(handler);
         }
         final shortcut = item.shortcut;
         if (shortcut != null) {
@@ -183,7 +182,7 @@ class MenuChannel {
         }
       } else {
         throw ArgumentError(
-            'Unknown AbstractMenuItem type: $item (${item.runtimeType})');
+            'Unknown AbstractNativeMenuItem type: $item (${item.runtimeType})');
       }
     }
     return representation;
@@ -191,12 +190,13 @@ class MenuChannel {
 
   /// Returns the representation of [menu] suitable for passing over the
   /// platform channel to the native plugin.
-  List<dynamic> _channelRepresentationForMenu(List<AbstractMenuItem> menu) {
+  List<dynamic> _channelRepresentationForMenu(
+      List<AbstractNativeMenuItem> menu) {
     final menuItemRepresentations = [];
     // Dividers are only allowed after non-divider items (see ApplicationMenu).
     var skipNextDivider = true;
     for (final menuItem in menu) {
-      final isDivider = menuItem is MenuDivider;
+      final isDivider = menuItem is NativeMenuDivider;
       if (isDivider && skipNextDivider) {
         continue;
       }
@@ -233,7 +233,8 @@ class MenuChannel {
         }
 
         if (key.keyLabel.isNotEmpty) {
-          channelRepresentation[_kShortcutKeyEquivalent] = key.keyLabel.toLowerCase();
+          channelRepresentation[_kShortcutKeyEquivalent] =
+              key.keyLabel.toLowerCase();
         } else {
           final specialKey = _shortcutSpecialKeyValues[key];
           if (specialKey == null) {
@@ -258,7 +259,7 @@ class MenuChannel {
   ///
   /// The returned ID should be attached to the menu so that the native plugin
   /// can identify the menu item selected in the callback.
-  int _storeMenuCallback(MenuSelectedCallback callback) {
+  int _storeMenuCallback(VoidCallback callback) {
     final id = _nextMenuItemId++;
     _selectionCallbacks[id] = callback;
     return id;

--- a/plugins/menubar/lib/src/native_menu_item.dart
+++ b/plugins/menubar/lib/src/native_menu_item.dart
@@ -13,36 +13,28 @@
 // limitations under the License.
 import 'package:flutter/widgets.dart';
 
-/// A callback provided to [MenuItem] to handle menu selection.
-typedef MenuSelectedCallback = void Function();
-
 /// The base type for an individual menu item that can be shown in a menu.
-abstract class AbstractMenuItem {
+abstract class AbstractNativeMenuItem {
   /// Creates a new menu item with the give label.
-  const AbstractMenuItem(this.label);
+  const AbstractNativeMenuItem(this.label);
 
   /// The displayed label for the menu item.
   final String label;
 }
 
 /// A standard menu item, with no submenus.
-class MenuItem extends AbstractMenuItem {
+class NativeMenuItem extends AbstractNativeMenuItem {
   /// Creates a new menu item with the given [label] and options.
-  ///
-  /// Note that onClicked should generally be set unless [enabled] is false,
-  /// or the menu item will be selectable but not do anything.
-  const MenuItem({
+  const NativeMenuItem({
     required String label,
     this.shortcut,
-    this.enabled = true,
-    this.onClicked,
+    this.onSelected,
   }) : super(label);
 
   /// The callback to call whenever the menu item is selected.
-  final MenuSelectedCallback? onClicked;
-
-  /// Whether or not the menu item is enabled.
-  final bool enabled;
+  ///
+  /// If null, the menu item is disabled.
+  final VoidCallback? onSelected;
 
   /// The shortcut/accelerator for the menu item, if any.
   ///
@@ -58,16 +50,17 @@ class MenuItem extends AbstractMenuItem {
 /// A menu item continaing a submenu.
 ///
 /// The item itself can't be selected, it just displays the submenu.
-class Submenu extends AbstractMenuItem {
+class NativeSubmenu extends AbstractNativeMenuItem {
   /// Creates a new submenu with the given [label] and [children].
-  const Submenu({required String label, required this.children}) : super(label);
+  const NativeSubmenu({required String label, required this.children})
+      : super(label);
 
   /// The menu items contained in the submenu.
-  final List<AbstractMenuItem> children;
+  final List<AbstractNativeMenuItem> children;
 }
 
 /// A menu item that serves as a divider, generally drawn as a line.
-class MenuDivider extends AbstractMenuItem {
+class NativeMenuDivider extends AbstractNativeMenuItem {
   /// Creates a new divider item.
-  const MenuDivider() : super('');
+  const NativeMenuDivider() : super('');
 }

--- a/plugins/menubar/lib/src/set_application_menu.dart
+++ b/plugins/menubar/lib/src/set_application_menu.dart
@@ -14,12 +14,12 @@
 import 'dart:async';
 
 import 'menu_channel.dart';
-import 'menu_item.dart';
+import 'native_menu_item.dart';
 
 /// Sets the application menu for the app based on the [menuSpec].
 ///
-/// Adjacent [MenuDivider]s will be coalesced, leading and/or trailing
-/// [MenuDivider]s will be removed.
-Future<Null> setApplicationMenu(List<Submenu> menuSpec) async {
+/// Adjacent [NativeMenuDivider]s will be coalesced, leading and/or trailing
+/// [NativeMenuDivider]s will be removed.
+Future<Null> setApplicationMenu(List<NativeSubmenu> menuSpec) async {
   await MenuChannel.instance.setMenu(menuSpec);
 }

--- a/plugins/menubar/pubspec.yaml
+++ b/plugins/menubar/pubspec.yaml
@@ -1,6 +1,6 @@
 name: menubar
 description: Provides the ability to add native menubar items with Dart callbacks.
-version: 0.1.0
+version: 0.2.0
 
 # Do not publish this plugin. See:
 # https://github.com/google/flutter-desktop-embedding/blob/master/plugins/README.md#using-plugins

--- a/testbed/lib/keyboard_test_page.dart
+++ b/testbed/lib/keyboard_test_page.dart
@@ -113,7 +113,7 @@ class _KeyboardTestPageState extends State<KeyboardTestPage> {
     setState(() {
       _messages.add(message);
     });
-    SchedulerBinding.instance?.addPostFrameCallback((_) {
+    SchedulerBinding.instance.addPostFrameCallback((_) {
       _scrollController.jumpTo(
         _scrollController.position.maxScrollExtent,
       );

--- a/testbed/lib/main.dart
+++ b/testbed/lib/main.dart
@@ -86,62 +86,66 @@ class _AppState extends State<MyApp> {
   /// Rebuilds the native menu bar based on the current state.
   void updateMenubar() {
     setApplicationMenu([
-      Submenu(label: 'Color', children: [
-        MenuItem(
+      NativeSubmenu(label: 'Color', children: [
+        NativeMenuItem(
             label: 'Reset',
-            enabled: _primaryColor != Colors.blue,
             shortcut: LogicalKeySet(
                 LogicalKeyboardKey.meta, LogicalKeyboardKey.backspace),
-            onClicked: () {
-              setPrimaryColor(Colors.blue);
-            }),
-        MenuDivider(),
-        Submenu(label: 'Presets', children: [
-          MenuItem(
+            onSelected: _primaryColor == Colors.blue
+                ? null
+                : () {
+                    setPrimaryColor(Colors.blue);
+                  }),
+        NativeMenuDivider(),
+        NativeSubmenu(label: 'Presets', children: [
+          NativeMenuItem(
               label: 'Red',
-              enabled: _primaryColor != Colors.red,
               shortcut: LogicalKeySet(LogicalKeyboardKey.meta,
                   LogicalKeyboardKey.shift, LogicalKeyboardKey.keyR),
-              onClicked: () {
-                setPrimaryColor(Colors.red);
-              }),
-          MenuItem(
+              onSelected: _primaryColor == Colors.red
+                  ? null
+                  : () {
+                      setPrimaryColor(Colors.red);
+                    }),
+          NativeMenuItem(
               label: 'Green',
-              enabled: _primaryColor != Colors.green,
               shortcut: LogicalKeySet(LogicalKeyboardKey.meta,
                   LogicalKeyboardKey.alt, LogicalKeyboardKey.keyG),
-              onClicked: () {
-                setPrimaryColor(Colors.green);
-              }),
-          MenuItem(
+              onSelected: _primaryColor == Colors.green
+                  ? null
+                  : () {
+                      setPrimaryColor(Colors.green);
+                    }),
+          NativeMenuItem(
               label: 'Purple',
-              enabled: _primaryColor != Colors.deepPurple,
               shortcut: LogicalKeySet(LogicalKeyboardKey.meta,
                   LogicalKeyboardKey.control, LogicalKeyboardKey.keyP),
-              onClicked: () {
-                setPrimaryColor(Colors.deepPurple);
-              }),
+              onSelected: _primaryColor == Colors.deepPurple
+                  ? null
+                  : () {
+                      setPrimaryColor(Colors.deepPurple);
+                    }),
         ])
       ]),
-      Submenu(label: 'Counter', children: [
-        MenuItem(
+      NativeSubmenu(label: 'Counter', children: [
+        NativeMenuItem(
             label: 'Reset',
-            enabled: _counter != 0,
             shortcut: LogicalKeySet(
                 LogicalKeyboardKey.meta, LogicalKeyboardKey.digit0),
-            onClicked: () {
-              _setCounter(0);
-            }),
-        MenuDivider(),
-        MenuItem(
+            onSelected: _counter == 0
+                ? null
+                : () {
+                    _setCounter(0);
+                  }),
+        NativeMenuDivider(),
+        NativeMenuItem(
             label: 'Increment',
             shortcut: LogicalKeySet(LogicalKeyboardKey.f2),
-            onClicked: incrementCounter),
-        MenuItem(
+            onSelected: incrementCounter),
+        NativeMenuItem(
             label: 'Decrement',
-            enabled: _counter > 0,
             shortcut: LogicalKeySet(LogicalKeyboardKey.f1),
-            onClicked: _decrementCounter),
+            onSelected: _counter == 0 ? null : _decrementCounter),
       ]),
     ]);
   }
@@ -151,12 +155,12 @@ class _AppState extends State<MyApp> {
     // Any time the state changes, the menu needs to be rebuilt.
     updateMenubar();
 
+    final theme = ThemeData();
     return MaterialApp(
       title: 'Flutter Demo',
-      theme: ThemeData(
-        primarySwatch: Colors.blue,
-        primaryColor: _primaryColor,
-        accentColor: _primaryColor,
+      theme: theme.copyWith(
+        colorScheme: theme.colorScheme
+            .copyWith(primary: _primaryColor, secondary: _primaryColor),
       ),
       darkTheme: ThemeData.dark(),
       home: _MyHomePage(title: 'Flutter Demo Home Page', counter: _counter),

--- a/testbed/linux/flutter/generated_plugin_registrant.cc
+++ b/testbed/linux/flutter/generated_plugin_registrant.cc
@@ -2,6 +2,8 @@
 //  Generated file. Do not edit.
 //
 
+// clang-format off
+
 #include "generated_plugin_registrant.h"
 
 #include <menubar/menubar_plugin.h>

--- a/testbed/linux/flutter/generated_plugin_registrant.h
+++ b/testbed/linux/flutter/generated_plugin_registrant.h
@@ -2,6 +2,8 @@
 //  Generated file. Do not edit.
 //
 
+// clang-format off
+
 #ifndef GENERATED_PLUGIN_REGISTRANT_
 #define GENERATED_PLUGIN_REGISTRANT_
 

--- a/testbed/linux/flutter/generated_plugins.cmake
+++ b/testbed/linux/flutter/generated_plugins.cmake
@@ -7,6 +7,9 @@ list(APPEND FLUTTER_PLUGIN_LIST
   window_size
 )
 
+list(APPEND FLUTTER_FFI_PLUGIN_LIST
+)
+
 set(PLUGIN_BUNDLED_LIBRARIES)
 
 foreach(plugin ${FLUTTER_PLUGIN_LIST})
@@ -15,3 +18,8 @@ foreach(plugin ${FLUTTER_PLUGIN_LIST})
   list(APPEND PLUGIN_BUNDLED_LIBRARIES $<TARGET_FILE:${plugin}_plugin>)
   list(APPEND PLUGIN_BUNDLED_LIBRARIES ${${plugin}_bundled_libraries})
 endforeach(plugin)
+
+foreach(ffi_plugin ${FLUTTER_FFI_PLUGIN_LIST})
+  add_subdirectory(flutter/ephemeral/.plugin_symlinks/${ffi_plugin}/linux plugins/${ffi_plugin})
+  list(APPEND PLUGIN_BUNDLED_LIBRARIES ${${ffi_plugin}_bundled_libraries})
+endforeach(ffi_plugin)

--- a/testbed/pubspec.yaml
+++ b/testbed/pubspec.yaml
@@ -3,7 +3,7 @@ description: A test project for flutter-desktop-embedding.
 
 environment:
   sdk: '>=2.12.0-0 <3.0.0'
-  flutter: '>=1.22.0-10.0.pre'
+  flutter: '>=3.0.0'
 
 dependencies:
   flutter:

--- a/testbed/windows/flutter/generated_plugin_registrant.cc
+++ b/testbed/windows/flutter/generated_plugin_registrant.cc
@@ -2,6 +2,8 @@
 //  Generated file. Do not edit.
 //
 
+// clang-format off
+
 #include "generated_plugin_registrant.h"
 
 #include <menubar/menubar_plugin.h>

--- a/testbed/windows/flutter/generated_plugin_registrant.h
+++ b/testbed/windows/flutter/generated_plugin_registrant.h
@@ -2,6 +2,8 @@
 //  Generated file. Do not edit.
 //
 
+// clang-format off
+
 #ifndef GENERATED_PLUGIN_REGISTRANT_
 #define GENERATED_PLUGIN_REGISTRANT_
 

--- a/testbed/windows/flutter/generated_plugins.cmake
+++ b/testbed/windows/flutter/generated_plugins.cmake
@@ -7,6 +7,9 @@ list(APPEND FLUTTER_PLUGIN_LIST
   window_size
 )
 
+list(APPEND FLUTTER_FFI_PLUGIN_LIST
+)
+
 set(PLUGIN_BUNDLED_LIBRARIES)
 
 foreach(plugin ${FLUTTER_PLUGIN_LIST})
@@ -15,3 +18,8 @@ foreach(plugin ${FLUTTER_PLUGIN_LIST})
   list(APPEND PLUGIN_BUNDLED_LIBRARIES $<TARGET_FILE:${plugin}_plugin>)
   list(APPEND PLUGIN_BUNDLED_LIBRARIES ${${plugin}_bundled_libraries})
 endforeach(plugin)
+
+foreach(ffi_plugin ${FLUTTER_FFI_PLUGIN_LIST})
+  add_subdirectory(flutter/ephemeral/.plugin_symlinks/${ffi_plugin}/windows plugins/${ffi_plugin})
+  list(APPEND PLUGIN_BUNDLED_LIBRARIES ${${ffi_plugin}_bundled_libraries})
+endforeach(ffi_plugin)

